qcodes/dataset/descriptions/versioning/converters.py

```diff
@@ -1,3 +1,15 @@
+"""
+This module contains functions which implement conversion between different
+(neighbouring) versions of RunDescriber.
+
+RunDescriber version log:
+
+- 0: The run_describer has a single attribute, interdeps, which is an instance
+of InterDependencies (which contains ParamSpecs)
+- 1: The run_describer has a single attribute, interdeps, which is an instance
+of InterDependencies_ (which contains ParamSpecBases)
+
```

Comment:
There are the in memory versions of the run describer? The serialization also has a version number? But that version is now as far as I can see version 1 which is still the old style serialization?

Reply:
correct, these are the in-memory versions. each version of RunDescriber implements its own/specific serialize/deserialize methods which define how that version is serialized/deserialized.

What we store on disk in this PR is still "version 0" structure - no change there (apart from the `"version": 0` field). What we use in qcodes in memory is version 1 - again as before this PR.